music_assistant/providers/internet-archive/provider.py

```diff
@@ -0,0 +1,794 @@
+        Uses heuristics to classify items as tracks, albums, or audiobooks.
+        """
+        identifier = doc.get("identifier")
+        if not identifier:
+            raise InvalidDataError("Missing identifier in search result")
+
+        title = clean_text(doc.get("title"))
+        creator = clean_text(doc.get("creator"))
+
+        # Be lenient - allow items without title if they have identifier
+        if not title and not identifier:
+            raise InvalidDataError("Missing both title and identifier in search result")
+
+        # Use identifier as fallback title if needed
+        if not title:
+            title = create_title_from_identifier(identifier)
+
+        # Determine what type of item this is
+        mediatype = doc.get("mediatype", "")
+        collection = doc.get("collection", [])
+        if isinstance(collection, str):
+            collection = [collection]
+
+        # Check if this is audiobook content
+        is_audiobook = any(coll in AUDIOBOOK_COLLECTIONS for coll in collection)
+
+        if is_audiobook and MediaType.AUDIOBOOK in media_types:
+            audiobook = doc_to_audiobook(
+                doc, self.domain, self.instance_id, self.client.get_item_url
+            )
+            if audiobook:
+                audiobooks.append(audiobook)
+
+        # For etree items, usually each item is an album (concert)
+        elif mediatype == "etree" or "etree" in collection:
+            if MediaType.ALBUM in media_types:
+                album = doc_to_album(doc, self.domain, self.instance_id, self.client.get_item_url)
+                if album:
+                    albums.append(album)
+
+            if MediaType.ARTIST in media_types and creator:
+                artist = create_artist(creator, self.domain, self.instance_id)
+                if artist and not artist_exists(artist, artists):
+                    artists.append(artist)
+
+        elif mediatype == "audio":
+            # Use heuristics to determine album vs track without expensive API calls
+            if is_likely_album(doc):
+                if MediaType.ALBUM in media_types:
+                    album = doc_to_album(
+                        doc, self.domain, self.instance_id, self.client.get_item_url
+                    )
+                    if album:
+                        albums.append(album)
+            elif MediaType.TRACK in media_types:
+                track = doc_to_track(doc, self.domain, self.instance_id, self.client.get_item_url)
+                if track:
+                    tracks.append(track)
+
+            if MediaType.ARTIST in media_types and creator:
+                artist = create_artist(creator, self.domain, self.instance_id)
+                if artist and not artist_exists(artist, artists):
+                    artists.append(artist)
+
+    @use_cache(expiration=86400 * 7)  # Cache for 1 week - artist "tracks" change infrequently
```

Comment:
```suggestion
    @use_cache(expiration=86400 * 60)  # Cache for 60 days - artist "tracks" change infrequently
```

Reply:
try to cache this static data as much as you can, it saves tons of requests to a provider